mechanics/physicell/src/config_reader.cpp

```diff
@@ -0,0 +1,299 @@
+#include "config_reader.h"
+
+#include <algorithm>
+#include <pugixml.hpp>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+
+namespace physicore::mechanics::physicell {
+
+namespace {
+
+pugi::xml_node get_required_child(const pugi::xml_node& parent, const char* name)
+{
+	pugi::xml_node child = parent.child(name);
+	if (!child)
+	{
+		throw std::runtime_error(std::string("Required XML tag <") + name + "> not found under <" + parent.name()
+								 + ">");
+	}
+	return child;
+}
+
+real_t parse_real(const pugi::xml_node& node, const char* tag_name)
+{
+	pugi::xml_node child = get_required_child(node, tag_name);
+	return static_cast<real_t>(child.text().as_double());
+}
+
+bool parse_bool(const pugi::xml_node& node, const char* tag_name)
+{
+	pugi::xml_node child = get_required_child(node, tag_name);
+	return child.text().as_bool();
+}
+
+template <typename Value>
+void ensure_size(std::vector<Value>& vec, std::size_t size, const Value& init)
+{
+	vec.assign(size, init);
+}
+
+template <typename Value>
+void ensure_matrix(std::vector<std::vector<Value>>& matrix, std::size_t rows, std::size_t cols, const Value& init)
+{
+	matrix.assign(rows, std::vector<Value>(cols, init));
+}
+
+} // namespace
+
+mechanics_config parse_simulation_parameters(const std::filesystem::path& config_file)
```

Comment:
this is way too big function - split it into smaller ones: general settings, seeking IDs, processing a cell definition (processing mechanics, motility)